### PR TITLE
Last try with all upgrades

### PR DESCRIPTION
**Pull request by stefaweb**
Seems ok.

Be careful. It was not easy to merge (first time for me) as I changed the layout and added items in many files.

### CODE REVIEW

File: mailscanner/login.php [hunk 1]
```diff
@@ -193,3 +193,4 @@
 
 </body>
 </html>
+r
```
Comment (endelwar) [inline mailscanner/login.php]: Is this an r char?

File: mailscanner/geoip_update.php [hunk 2]
```diff
@@ -221,4 +223,4 @@
 // Add the footer
 html_end();
 // close the connection to the Database
-dbclose();
+dbclose();
```
Comment (endelwar) [inline mailscanner/geoip_update.php]: please leave a new line at the end of file

File: mailscanner/functions.php [hunk 3]
```diff
@@ -717,9 +717,16 @@ function html_end($footer = "")
...
-    echo '<p class="center" style="font-size:13px"><i>' . "\n";
-    page_creation_timer();
-    echo '</i></p>' . "\n";
+    if (DEBUG) {
+       echo '<p class="center" style="font-size:13px"><i>' . "\n";
+       page_creation_timer();
+       echo '</i></p>' . "\n";
+    }
+    echo '<p class="center" style="font-size:13px">' . "\n";
+    echo 'MailWatch for MailScanner v';
```
Comment (endelwar) [inline mailscanner/functions.php]: This should be translated

File: mailscanner/other.php [hunk 4]
```diff
@@ -43,6 +43,7 @@
 html_start('Tools', "0", false, false);
 
 echo '<table width="100%" class="boxtable">
+ <tr><th>Tools/Links</th></tr>
```
Comment (endelwar) [inline mailscanner/other.php]: This should be translated

File: mailscanner/quarantine.php [hunk 5]
```diff
@@ -66,8 +67,8 @@
         if (count($items) > 0) {
             // Sort in reverse chronological order
             arsort($items);
-            echo '<table class="mail" cellspacing="1" align="center">' . "\n";
-            echo '<tr><th>Folder</th></tr>' . "\n";
+            echo '<table class="mail" width="100%" cellspacing="2" align="center">' . "\n";
+            echo '<tr><th colspan=2>Quarantine Folder</th></tr>' . "\n";
```
Comment (endelwar) [inline mailscanner/quarantine.php]: This should be translated

File: mailscanner/sa_rules_update.php [hunk 6]
```diff
@@ -43,18 +43,19 @@
...
-    echo "   This utility should generally be run after a SpamAssassin update, however it is safe to run at any time as it only replaces the existing values and inserts only new values in the table (therefore preserving descriptions from potentially deprecated or removed rules).<br>";
+    echo "   This utility should generally be run after a SpamAssassin update, however it is safe to run at any time as it only replaces the existing values and inserts only new values in the table (therefore preserving descriptions from potentially deprecated or removed rules).<br><br>";
     echo "  </td>";
     echo "</tr>";
     echo " <tr>";
     echo "  <td align=\"center\">
     <form method=\"post\" action=\"" . sanitizeInput($_SERVER['PHP_SELF']) . "\">
     <div style=\"margin: 5px\">
-    <input type=\"submit\" value=\"run now\">
+    <input type=\"submit\" value=\"Run Now\"><br><br>
```
Comment (endelwar) [inline mailscanner/sa_rules_update.php]: This should be translated

File: mailscanner/sf_version.php [hunk 7]
```diff
@@ -58,31 +58,41 @@
...
     echo '<table width="100%" class="boxtable">' . "\n";
+    echo '<tr><th>Software Versions</th></tr>' . "\n";
     echo '<tr>' . "\n";
     echo '<td>' . "\n";
 
-    echo '<p class="center" style="font-size:20px"><b>' . __('softver11') . '</b></p>' . "\n";
-    echo 'MailWatch ' . __('version11') . ' = ' . $mailwatch_version . '<br>' . "\n";
     echo '<br>' . "\n";
-    echo 'MailScanner ' . __('version11') . ' = ' . $mailscanner_version . '<br>' . "\n";
+    echo 'MailWatch version: ' . $mailwatch_version . '<br>' . "\n";
```
Comment (endelwar) [inline mailscanner/sf_version.php]: This should be translated

File: mailscanner/sf_version.php [hunk 8]
```diff
@@ -58,31 +58,41 @@
...
     echo '<br>' . "\n";
-    echo 'PHP ' . __('version11') . ' = ' . $php_version . '<br>' . "\n";
+    echo 'MailScanner ' . __('version11') . $mailscanner_version . '<br>' . "\n";
+    echo '<br>';
+    echo 'ClamAV ' . __('version11');
```
Comment (endelwar) [inline mailscanner/sf_version.php]: Not everyone uses clamav, there should be a check here (and also there should be no antivirus discrimination ;) )

File: mailscanner/sf_version.php [hunk 9]
```diff
@@ -58,31 +58,41 @@
...
+    echo '<br>' . "\n";
+    echo 'GeoIP Database ' . __('version11');
     if (FALSE != $geoipv4_version) {
         echo $geoipv4_version;
     } else {
         echo __('nodbdown11');
     }
     echo "<br>\n<br>\n";
-    echo 'GeoIPv6 Database ' . __('version11') . ' = ';
+    echo 'GeoIP Database ' . __('version11');
```
Comment (endelwar) [inline mailscanner/sf_version.php]: keep GeoIPV6, these are two separate databases

File: CHANGELOG.md [hunk 10]
```diff
@@ -18,6 +18,16 @@
  - Fix virus count sorting on "Virus Report"
  - Remove additional slashes in "SpamAssassin Rule Description Update"
  - Adding translation to user interface
+ - Add LOGO path in conf.php.example and changes done in corresponding files.
+ - layout changes in quarantine.php, other.php, sf_version.php
+ - Removed duplicate PHP function in tools/Cron_jobs/quarantine_report.php (is in functions.php)
+ - Changes in tools/Sendmail_relay/INSTALL
+ - Changes in tools/Sendmail_relay/sendmail_relay.init
```
Comment (endelwar) [inline CHANGELOG.md]: some more info?

File: mailscanner/geoip_update.php [hunk 11]
```diff
@@ -47,8 +47,10 @@
     echo '<form method="POST" action="' . sanitizeInput($_SERVER['PHP_SELF']) . '">
 	 <input type="hidden" name="run" value="true">
 	 <table class="boxtable" width="100%">
+            <tr><th>Update GeoIP Database</th></tr>
```
Comment (endelwar) [inline mailscanner/geoip_update.php]: This should be translated

File: mailscanner/login.php (hunk 1 above)
Comment (stefaweb) [inline mailscanner/login.php]: keyboard type by error
remove

File: mailscanner/sf_version.php (hunk 8 above)
Comment (stefaweb) [inline mailscanner/sf_version.php]: go on! ;-)

A ":" should be added to version11 in loc files.

'version11' => 'version:',

**Comment by stefaweb** [comment]
Do I need to do something?
Or do you take care of the mods?

File: mailscanner/sf_version.php (hunk 9 above)
Comment (stefaweb) [inline mailscanner/sf_version.php]: wrong copy/paste

File: CHANGELOG.md (hunk 10 above)
Comment (stefaweb) [inline CHANGELOG.md]: Some seems missing.

 - Add LOGO path in conf.php.example and changes done in corresponding files.
 - layout changes in quarantine.php, other.php, sf_version.php
 - Removed duplicate PHP function in tools/Cron_jobs/quarantine_report.php (is in functions.php)
 - Changes in tools/Sendmail_relay/INSTALL
 - Changes in tools/Sendmail_relay/sendmail_relay.init
 - Background color change in login.php
 - Add clamav and spamassassin version in sf_version.php
 - Changes in INSTALL and UPGRADING
 - Modified footer function in functions.php (page footer and DEBUG for page_creation_timer)
 - Add a check for 'subtests=' to add space after comma (to fit the screen) in sa_lint.php

Comment (endelwar) [inline CHANGELOG.md]: I mean: could you explain what changes has be done in tools/Sendmail_relay/INSTALL and tools/Sendmail_relay/sendmail_relay.init, so that the casual reader/installer know if he has to take action or not.

**Comment by endelwar** [comment]
Please fix the thing that I noted in your branch and push it to github

File: CHANGELOG.md (hunk 10 above)
Comment (stefaweb) [inline CHANGELOG.md]: I removed rc.d in  /etc/rc.d/init.d/ and .init

Uptodate for Debian and Ubuntu.

The last version for tools/Sendmail_relay/INSTALL:

# cp tools/Sendmail_relay/sendmail_relay.php /usr/local/bin/.
# cp tools/Sendmail_relay/sendmail_relay.init /etc/init.d/.
# chmod +x /etc/init.d/sendmail_relay
# /etc/init.d/sendmail_relay.init start
# update-rc.d sendmail_relay.init defaults

**Comment by stefaweb** [comment]
What is happening again?

**Comment by stefaweb** [comment]
over

**Comment by stefaweb** [comment]
Added version in all languages files.

**Comment by stefaweb** [comment]
@endelwar 

Seems ok?

File: mailscanner/conf.php.example.orig [hunk 12]
```diff
@@ -0,0 +1,247 @@
+<?php
```
Comment (endelwar) [inline mailscanner/conf.php.example.orig]: Why is this file here?

Comment (stefaweb) [inline mailscanner/conf.php.example.orig]: delete

**Comment by stefaweb** [comment]
I don't know how I've done this.
conf.php.example.orig

File: mailscanner/languages/de.php [hunk 13]
```diff
@@ -32,199 +32,205 @@
...
-    'pontspam12' => 'Spam Bewertung:',
-    'hpontspam12' => 'Hohe Spam Bewertung:',
-    'usedefault12' => 'Voreinstellungen benutzen',
-    'action_0212' => 'Aktion:',
-    'reset12' => 'Zurücksetzen',
-    'areusuredel12' => 'Soll der Benutzer tatsächlich gelöscht werden',
-    'errorpass12' => 'Passwörter stimmen nicht überein',
-
-    'i18_missing' => 'Keine deutsche Übersetzung',
+	// 01-login.php
```
Comment (endelwar) [inline mailscanner/languages/de.php]: please don't replace the spaces with tabs, I would like to keep PSR compatibility (even with all the limits that the actual code has)

Comment (stefaweb) [inline mailscanner/languages/de.php]: I was not aware of that.

What is PSR?

Comment (endelwar) [inline mailscanner/languages/de.php]: PSR are a series of guidelines for php developer, look at [PSR-1](http://www.php-fig.org/psr/psr-1/) and [PSR-2](http://www.php-fig.org/psr/psr-2/) for coding style

File: mailscanner/conf.php.example.orig (hunk 12 above)
Comment (endelwar) [inline mailscanner/conf.php.example.orig]: This file was created when you merged the master branch, which was in conflict with some editing you've done, git advise you when this happen and you must fix this conflicts before committing the new code, welcome to git ;).

**Comment by stefaweb** [comment]
Sorry but I'v no access to the live server. Typing without testing.

**Comment by stefaweb** [comment]
Enfin !!!

Off now.

**Comment by endelwar** [comment]
good work!
